File: Day03_ProgressDialog_SMS/app/src/main/java/day01/nguyenhongphat0/day03_progressdialog_sms/SendSMSActivity.java
package day01.nguyenhongphat0.day03_progressdialog_sms;

import android.app.Activity;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.telephony.SmsManager;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class SendSMSActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_send_sms);
    }

    public void clickToSend(View view) {
        EditText phone = findViewById(R.id.txtPhone);
        EditText content = findViewById(R.id.txtContent);
        final SmsManager sms = SmsManager.getDefault();
        Intent intent = new Intent("ACTION_MSG_SENT");
        final PendingIntent pendingIntent = PendingIntent.getBroadcast(this, 0, intent, 0);
        registerReceiver(new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                int result = getResultCode();
                String msg = "Sent OK";
                if (result != Activity.RESULT_OK) {
                    msg = "Sent Failed";
                }
                Toast.makeText(SendSMSActivity.this, msg, Toast.LENGTH_LONG).show();
            }
        }, new IntentFilter("ACTION_MSG_SENT"));
        sms.sendTextMessage(phone.getText().toString(), null, content.getText().toString(), pendingIntent, null);
        finish();
    }
}

File: kotlin/Day03/app/src/main/java/day03/nguyenhongphat0/day03/SendSMSActivity.kt
package day03.nguyenhongphat0.day03

import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity;
import android.telephony.SmsManager
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import kotlinx.android.synthetic.main.activity_send_sms.*
import kotlinx.android.synthetic.main.content_send_sms.*
import java.lang.Exception

class SendSMSActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_send_sms)
        setSupportActionBar(toolbar)

        fab.setOnClickListener {
            val phone = findViewById<EditText>(R.id.phoneText).text.toString()
            val content = findViewById<EditText>(R.id.contentText).text.toString()
            val pendingIntent = PendingIntent.getBroadcast(this, 0, Intent("SMS_SENT"), 0)
            val broadcastReceiver = object : BroadcastReceiver() {
                override fun onReceive(context: Context?, intent: Intent?) {
                    Toast.makeText(this@SendSMSActivity, "Result code: " + resultCode, Toast.LENGTH_LONG).show()
                }
            }
            registerReceiver(broadcastReceiver, IntentFilter("SMS_SENT"))
            try {
                val sms = SmsManager.getDefault()
                sms.sendTextMessage(phone, null, content, pendingIntent, pendingIntent)
            } catch (e: Exception) {
                Toast.makeText(this, e.message, Toast.LENGTH_LONG).show()
                e.printStackTrace()
            }
            finish()
        }
    }

}
